Lib/idlelib/idle_test/test_config.py

```diff
@@ -185,6 +186,358 @@ def test_save(self):
             self.assertFalse(os.path.exists(path))
 
 
+class IdleConfTest(unittest.TestCase):
+    """Test for idleConf"""
+
+    def new_config(self, _utest=False):
+        return config.IdleConf(_utest=_utest)
+
+    def mock_config(self):
+        """Return a mocked idleConf
+
+        Both default and user config used the same config-*.def
+        """
+        import sys
+        conf = config.IdleConf(_utest=True)
+        if __name__ != '__main__':
+            idle_dir = os.path.dirname(__file__)
+        else:
+            idle_dir = os.path.abspath(sys.path[0])
+        for ctype in conf.config_types:
+            config_path = os.path.join(idle_dir, '../config-%s.def' % ctype)
+            conf.defaultCfg[ctype] = config.IdleConfParser(config_path)
+            conf.userCfg[ctype] = config.IdleUserConfParser(config_path)
+        conf.LoadCfgFiles()
+
+        return conf
+
+    def test_get_user_cfg_dir(self):
```

Comment:
I think this can be reference from ipython source code, it turyly seperate this function from a class like what we do here. https://github.com/ipython/ipython/blob/master/IPython/utils/path.py#L172

I think this will need to be discuss on mailing list or bpo. But I prefer to first finish the unittest, then do other stuff to change the behavior.